app/controllers/bulk_submissions_controller.rb

```diff
@@ -10,6 +10,19 @@ def destroy
     redirect_back(fallback_location: authenticated_root_path)
   end
 
+  def cancel
+    bulk_submission = BulkSubmission.find(params[:id])
+    bulk_submission.original_file.purge
+    bulk_submission.destroy!
+    redirect_back(fallback_location: authenticated_root_path)
+  end
+
+  def discard
+    bulk_submission = BulkSubmission.find(params[:id])
+    bulk_submission.discard
```

Comment:
use bang to fail noisily
```suggestion
    bulk_submission.discard!
```